make_ghpages/make_pages.py

```diff
@@ -80,158 +79,154 @@ def get_git_branches(git_url):
-            meta_info = get_meta_info(app_data['meta_url'])
-        else:
-            raise exc.MissingMetadata("No 'meta_url' key for '{}' in apps.json".format(app_name))
-
-        # Check if categories are specified, warn if not
-        if 'categories' not in app_data:
-            print("  >> WARNING: No categories specified.")
-        # Check correct categories are specified
-        else:
-            validate_categories(app_data['categories'], all_data['categories'])
-
-        app_data['metainfo'] = validate_meta_info(app_name, meta_info, app_data['git_url'])
-        app_data['gitinfo'] = get_git_branches(app_data['git_url'])
-        app_data['subpage'] = subpage_name
-        app_data['hosted_on'] = hosted_on
-        
-        # Get logo URL, if it has been specified
-        if 'logo' in app_data['metainfo']:
-            app_data['logo'] = get_logo_url(app_data['metainfo']['logo'], app_data['meta_url'])
-
-        all_data['apps'][app_name] = app_data
-
-        # Make single-entry page based on singlepage.html
-        app_html = singlepage_template.render(category_map=all_data['categories'], **app_data)
+        app_html = singlepage_template.render(category_map=apps_meta['categories'], **app_data)
         with codecs.open(subpage_abspath, 'w', 'utf-8') as f:
             f.write(app_html)
-        print("    - Page {} generated.".format(subpage_name))
+        print(f"  - {subpage_name}")
 
     # Make index page based on main_index.html
     print("[main index]")
-    rendered = main_index_template.render(**all_data)
-    outfile = os.path.join(outdir_abs, 'index.html')
-    with codecs.open(outfile, 'w', 'utf-8') as f:
-        f.write(rendered)
-    print("  - index.html generated")
+    rendered = main_index_template.render(**apps_meta)
+    outfile = BUILD_PATH / 'index.html'
+    outfile.write_text(rendered, encoding='utf-8')
+    print(f"  - {outfile.relative_to(BUILD_PATH)}")
 
     # Save json data for the app manager
-    outfile = os.path.join(outdir_abs, apps_meta_file)
-    with codecs.open(outfile, 'w', 'utf-8') as f:
-        json.dump(all_data, f, ensure_ascii=False, indent=2)
-    print("  - apps_meta.json generated")
+    outfile = BUILD_PATH / 'apps_meta.json'
+    rendered = json.dumps(apps_meta, ensure_ascii=False, indent=2)
+    outfile.write_text(rendered, encoding='utf-8')
+    print(f"  - {outfile.relative_to(BUILD_PATH)}")
 
     # Copy schemas
     print("[schemas/v1]")
-    schemas_outdir = Path(outdir_abs).joinpath('schemas/v1')
+    schemas_outdir = BUILD_PATH.joinpath('schemas', 'v1')
     schemas_outdir.mkdir(parents=True)
-    for schemafile in Path(pwd).glob('../schemas/*.schema.json'):
-        print(f"  - {schemafile.name}")
+    for schemafile in ROOT.glob('schemas/*.schema.json'):
         shutil.copyfile(schemafile, schemas_outdir.joinpath(schemafile.name))
+        print(f"  - {schemas_outdir.relative_to(BUILD_PATH)}/{schemafile.name}")
+
+
+if __name__ == "__main__":
```

Comment:
Just to keep to either double or single quotation strings.
```suggestion
if __name__ == '__main__':
```